### PR TITLE
Next Auth Adapter callback functions

### PR DESCRIPTION
**Pull request by GloireMutaliko21**
This PR completes also this #2563 and add some other features and fixes

### CODE REVIEW

File: apps/web/app/[locale]/auth/error/page.tsx [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
 		<div className="mt-28 flex flex-col gap-7 items-center">
 			<div className="m-auto relative flex justify-center items-center gap-4 text-center ">
 				<SadCry width={97} height={97} />
-				<Text className="text-[78px] font-semibold text-chetwodeBlue">{t('pages.error.TITLE')}</Text>
+				<Text className="text-[78px] font-semibold text-chetwodeBlue">Error !</Text>
 			</div>
 
 			<Text className="text-[40px] font-bold text-center text-[#282048] dark:text-light--theme">
-				{t('pages.error.HEADING_TITLE')}
+				Something went wrong !
```
Comment (evereq) [inline apps/web/app/[locale]/auth/error/page.tsx:23]: @GloireMutaliko21 why you remove translations!?

File: apps/web/app/services/server/requests/OAuth.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+};
+
+async function signIn(provider: ProviderEnum, access_token: string) {
+	try {
+		const gauzyUser = await signWithSocialLoginsRequest(provider, access_token);
+
+		if (!gauzyUser) {
+			return Promise.reject({
+				errors: {
+					email: 'Your account is not yet ready to be used on the Ever Teams Platform'
```
Comment (evereq) [inline apps/web/app/services/server/requests/OAuth.ts:23]: @GloireMutaliko21 best if you use translation in such error mesages

**Comment by evereq** [comment]
@GloireMutaliko21 please fix all my comments in NEXT PR

File: apps/web/app/[locale]/auth/error/page.tsx (hunk 1 above)
Comment (GloireMutaliko21) [reply apps/web/app/[locale]/auth/error/page.tsx:23]: Ohhh ! Will fix it in next PR...

**Pull request merged**